### PR TITLE
Add reference documentation for Kustomize in Kubectl

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: docs/book/pages/reference/kustomize.md [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+| :--------------------------------- | :--------------- | |
+| [bases](#bases)                    | Provider         | Add Resource Configs from another `kustomization.yaml` |
+| [commonAnnotations](#commonannotations)   | Transformer         | Set annotations on all Resources and Selectors. |
+| [commonLabels](#commonlabels)      | Transformer         | Set labels on all Resources and Selectors. |
+| [configMapGenerator](#configmapgenerator)                    | Provider         | Generate ConfigMap Resources. |
+| [configurations](#configurations)                    | Meta         | Manually configure how Transformers are applied to CRDs. |
+| [crds](#crds)                    | Meta         | Automatically configure how Transformers are applied to CRDs. |
+| [images](#images)                    | Transformer         | Override image names and tags| 
+| [namespace](#namespace)                    | Transformer         | Override namespaces on all Resources. |
+| [namePrefix](#namePrefix)                    | Transformer         | Override names with a name prefix for all Resources and References. |
```
Comment (monopole) [inline docs/book/pages/reference/kustomize.md]: Add a prefix to the names of all ...

File: docs/book/pages/reference/kustomize.md [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+| [nameSuffix](#nameSuffix)                    | Transformer         | Override names with a name suffix for all Resources and References. |
```
Comment (monopole) [inline docs/book/pages/reference/kustomize.md]: ditto

**Comment by pwittrock** [comment]
PTAL

File: docs/book/pages/reference/kustomize.md (hunk 2 above)
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

File: docs/book/pages/reference/kustomize.md (hunk 1 above)
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

File: docs/book/pages/reference/kustomize.md [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+
+## Table of Contents
+
+| Name                               | Type             | Descriptions |
+| :--------------------------------- | :--------------- | |
+| [bases](#bases)                    | Provider         | Add Resource Configs from another `kustomization.yaml` |
+| [commonAnnotations](#commonannotations)   | Transformer         | Set annotations on all Resources and Selectors. |
+| [commonLabels](#commonlabels)      | Transformer         | Set labels on all Resources and Selectors. |
+| [configMapGenerator](#configmapgenerator)                    | Provider         | Generate ConfigMap Resources. |
+| [configurations](#configurations)                    | Meta         | Manually configure how Transformers are applied to CRDs. |
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

File: docs/book/pages/reference/kustomize.md [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+| [crds](#crds)                    | Meta         | Automatically configure how Transformers are applied to CRDs. |
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

File: docs/book/pages/reference/kustomize.md [hunk 5]
```diff
@@ -0,0 +1,600 @@
+{% panel style="info", title="TL;DR" %}
+- Reference for `kustomization.yaml`
+{% endpanel %}
+
+# Kustomization.yaml Reference
+
+#### Terms:
+
+- **Providers**: Provide Resource Config to Kustomize to be transformed.
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

File: docs/book/pages/reference/kustomize.md [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+
+{% method %}
+
+This field contains a list of paths to **directories or git repositories** containing `kustomization.yaml`s.
+
+`bases` produce Resource Config by running Kustomize against the target.  The result will then have Tranformers from
+the current `kustomization.yaml` applied.
+
+`bases` work similarly to a base layer in a Docker images - e.g. inherit everything from this base, and then apply
+changse to it.
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply docs/book/pages/reference/kustomize.md]: done

**Pull request merged**